controllers/gce/utils/utils_test.go

```diff
@@ -0,0 +1,482 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"fmt"
+	compute "google.golang.org/api/compute/v1"
+	"google.golang.org/api/googleapi"
+	"testing"
+)
+
+func TestGetSetClusterName(t *testing.T) {
+	fooTests := []struct {
+		cn  string
+		ecn string
+	}{
+		{"", ""},
+		{"fooCN", "fooCN"},
+		{"fooCN01--fooCN02--fooCN03", "fooCN03"},
+	}
+
+	namer := &Namer{}
+	for _, fooTest := range fooTests {
+		namer.SetClusterName(fooTest.cn)
+		rcn := namer.GetClusterName()
+		if rcn != fooTest.ecn {
+			t.Errorf("returned %v but expected %v", rcn, fooTest.ecn)
+		}
+	}
+}
+
+func TestGetSetFireName(t *testing.T) {
+	fooTests := []struct {
+		fn  string
+		efn string
+	}{
+		{"", ""},
+		{"fooFN", "fooFN"},
+		{"fooFN01--fooFN02--fooFN03", "fooFN01--fooFN02--fooFN03"},
+	}
+
+	namer := &Namer{}
+	for _, fooTest := range fooTests {
+		namer.SetFirewallName(fooTest.fn)
+		rfn := namer.GetFirewallName()
+		if rfn != fooTest.efn {
+			t.Errorf("returned %v but expected %v", rfn, fooTest.efn)
+		}
+	}
+}
+
+func TestNerNamer(t *testing.T) {
```

Comment:
This test seems like overkill considering you already test the sets and gets of each name type.  I'd be satisfied with a single, simple test. 
```go 
cn, fn := "my-cluster-name", "my-firewall-name"
n := NewNamer(cn, fn)
if n.clusterName != cn || n.firewallName != fn {
    ....
}
```